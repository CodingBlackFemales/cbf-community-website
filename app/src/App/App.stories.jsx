import React from 'react';

import { storiesOf } from '@storybook/react';

import App from './';

storiesOf('Main Components', module).add('App', () => <App />);
