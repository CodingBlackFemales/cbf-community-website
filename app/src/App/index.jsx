import React from 'react';
import styled from 'styled-components';

const App = ({ className }) => {
  const classPrefix = 'app';
  return (
    <div className={(classPrefix, className)}>
      <header className={`${classPrefix}__header`}>
        <p>Placeholder text</p>
      </header>
    </div>
  );
};

const StyledApp = styled(App)`
  text-align: center;

  .app__header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }
`;

export default StyledApp;
