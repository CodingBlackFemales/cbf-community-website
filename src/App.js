import React, { Component } from "react";
import "./App.css";
import Theme from "./Theme";

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <Theme />
      </React.Fragment>
    );
  }
}

export default App;
