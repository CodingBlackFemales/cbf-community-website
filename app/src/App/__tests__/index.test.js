import React from 'react';
import { cleanup } from '@testing-library/react';

import App from '..';
import {
  mountWithTheme,
  renderWithTheme,
} from '../../shared/utils/test/testHelpers';

describe('<App', () => {
  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { asFragment } = renderWithTheme(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the header text', () => {
    const appComponent = mountWithTheme(<App />);

    const header = appComponent.find('.app__header');

    expect(header.exists()).toBe(true);
  });
});
