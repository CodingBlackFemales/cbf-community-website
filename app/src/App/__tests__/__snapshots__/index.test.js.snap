// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c0 .app__header {
  background-color: #282c34;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

<div
    class="c0"
  >
    <header
      class="app__header"
    >
      <p>
        Placeholder text
      </p>
    </header>
  </div>
</DocumentFragment>
`;
